bundles/org.openhab.binding.teleinfo/src/main/java/org/openhab/binding/teleinfo/internal/serial/TeleinfoSerialControllerHandler.java

```diff
@@ -0,0 +1,199 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.teleinfo.internal.serial;
+
+import static org.openhab.binding.teleinfo.internal.TeleinfoBindingConstants.*;
+
+import java.io.IOException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.io.transport.serial.PortInUseException;
+import org.eclipse.smarthome.io.transport.serial.SerialPort;
+import org.eclipse.smarthome.io.transport.serial.SerialPortIdentifier;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.eclipse.smarthome.io.transport.serial.UnsupportedCommOperationException;
+import org.openhab.binding.teleinfo.internal.dto.Frame;
+import org.openhab.binding.teleinfo.internal.handler.TeleinfoAbstractControllerHandler;
+import org.openhab.binding.teleinfo.internal.reader.io.serialport.InvalidFrameException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TeleinfoSerialControllerHandler} class defines a handler for serial controller.
+ *
+ * @author Nicolas SIBERIL - Initial contribution
+ */
+@NonNullByDefault
+public class TeleinfoSerialControllerHandler extends TeleinfoAbstractControllerHandler
+        implements TeleinfoReceiveThreadListener {
+
+    private final Logger logger = LoggerFactory.getLogger(TeleinfoSerialControllerHandler.class);
+
+    private static final int SERIAL_RECEIVE_TIMEOUT = 250;
+    private static final int SERIAL_PORT_DELAY_RETRY_IN_SECONDS = 60;
+
+    private SerialPortManager serialPortManager;
+    private @Nullable SerialPort serialPort;
+    private @Nullable TeleinfoReceiveThread receiveThread;
+    private @Nullable ScheduledFuture<?> keepAliveThread;
+    private long invalidFrameCounter = 0;
+
+    public TeleinfoSerialControllerHandler(Bridge thing, SerialPortManager serialPortManager) {
+        super(thing);
+        this.serialPortManager = serialPortManager;
+    }
+
+    @Override
+    public void initialize() {
+        invalidFrameCounter = 0;
+
+        keepAliveThread = scheduler.scheduleWithFixedDelay(() -> {
+            if (!isInitialized()) {
+                updateStatus(ThingStatus.UNKNOWN);
```

Comment:
After the thing status is set to unknown, the framework starts to work with this Thing: `handleCommand()` and `dispose()` can be invoked (concurrently). When `dispose()` is invoked concurrently in this method, before the receiving thread has been created, it won't be stopped. The status should be set to unknown as one of the last actions.